Source/Orts.Simulation/Common/Scripting/ScriptManager.cs

```diff
@@ -34,7 +34,8 @@ public class ScriptManager
     {
         readonly Simulator Simulator;
         readonly Dictionary<string, Assembly> Scripts = new Dictionary<string, Assembly>();
-        static readonly CSharpCodeProvider Compiler = new CSharpCodeProvider();
+        static readonly ProviderOptions ProviderOptions = new ProviderOptions("roslyn/csc.exe", 10);
```

Comment:
This, to me, is a blocker - you're assuming that the working directory will always be the directory with RunActivity.exe, but this will not always be the case, particularly if you're running OR from the command line or through a debugger.

You should explicitly request the full path to the Open Rails executable. For reference, [this](https://github.com/openrails/openrails/blob/4d61887252746f5adc2714d9f7950d7a4fe8a097/Source/RunActivity/Viewer3D/Processes/Game.cs#L95) is how it's done for the Content\ directory. Since the Orts.Simulation project lacks a reference to Windows.Forms, I suggest storing this in that very same `Game()` constructor and passing it to the `Simulator` class, but it's up to you.

Reply:
Alright, this is fixed with the new commit.